### DISCUSSION TITLE
2.0.0

### DISCUSSION THREAD

**Discussion by feanor-anglin** [GoWired/GoWired-Project#32] [General]
This release introduces some important changes to GoWired software. From now on the software will require a GoWired-lib library added to Arduino IDE to compile. GoWired-lib is available via the Arduino Library Manager and installs with most GoWired dependencies (3rd party libraries this software uses). Other changes:
- The code has been rebuilt and unified to use the same GoWired-lib library files
- GoWired Touch software is now also available in Software directory of GoWired-Project repository
- New sensor type: Configuration Sensor, which enables configuration options by message or through notify.mysensors Home Assistant service
- Roller shutter calibration can now be called out using message / notify.mysensors service
- 8RelayDin software is now fixed, improved and ready to use (operation with security sensors, possibility of inverting input logic, possibility of declaring inputs & outputs as independent entities)
- Improved shutters / blinds operation procedure
- Improved reading inputs procedure
- Various improvements & bugfixes

<hr /><em>This discussion was created from the release <a href='https://github.com/GoWired/GoWired-Project/releases/tag/2.0.0'>2.0.0</a>.</em>

**Comment by djvl** [comment]
Congratulations for achieving this milestone! Thanks for all the effort that has gone into it. Will soon start to upgrade my GoWired setup and will report back.

**Reply by feanor-anglin** [reply]
Thank you for your support and enthusiasm! Please let us know about the outcome of your upgrade.

**Reply by djvl** [reply]
Over the last couple of days I upgraded all (3x 2ssr & 6x rgbw) modules to the 2.0 version without major issues. The only thing that took some more time was the Gateway-MQTT. Had some weird behavior (packet loss and disconnects) with a new version of the UIPEthernet library. This was ultimately fixed using the exact same version (2.0.7) as you have in your personal repository.
Another thing I want to mention is the time consuming process of upgrading firmware per module. For me this is still doable, but when your setup grows it would be nice to do this remotely (OTA). Is this on the roadmap?

**Reply by feanor-anglin** [reply]
Thank you for your feedback. It is worth noticing that new versions of UIPEthernet cause problems.

Referring to FOTA, there is such a functionality available in MySensors protocol. But it requires some knowledge and preparations (most of all, sensebender bootloader). We actively use it though. [Here](https://www.mysensors.org/about/fota) you have a complete guide.

And yes, we have simplifying this process in mind.
